sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -197,7 +198,14 @@ class QueryExecution(
         queryExecution.toString(maxFields, append)
       case CodegenMode =>
         try {
-          org.apache.spark.sql.execution.debug.writeCodegen(append, queryExecution.executedPlan)
+          queryExecution.executedPlan match {
```

Comment:
How about handling the AQE node in the `writeCodegen` side instead? That way, `debugCodegen` can support it, too?

Reply:
@maropu - updated, thanks.